jextract generated code watnings: try must be before await

```
/Users/ktoso/code/swift-java/Samples/SwiftJavaExtractJNISampleApp/.build/plugins/outputs/swiftjavaextractjnisampleapp/MySwiftLibrary/destination/JExtractSwiftPlugin/Sources/MySwiftLibraryModule+SwiftJava.swift:472:28: warning: 'try' must precede 'await'
470 |       } // printTaskBody(printer:) @ JExtractSwiftLib/JNISwift2JavaGenerator+NativeTranslation.swift:976
471 |       do {
472 |         let swiftResult$ = await try MySwiftLibrary.asyncThrows()
    |                            `- warning: 'try' must precede 'await'
473 |         environment = try! JavaVirtualMachine.shared().environment()
```

We need to fix those, we're getting flooded with warnings now

--- a/Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+NativeTranslation.swift
+++ b/Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+NativeTranslation.swift
@@ -1033,15 +1033,21 @@ extension JNISwift2JavaGenerator {
         )
 
         func printDo(printer: inout CodePrinter) {
-          // Generate proper 'try await' order: if placeholder already has 'try', remove it and add 'try await', otherwise just add 'try await'
-          let modifiersAndPlaceholder: String
-          if placeholder.hasPrefix("try ") {
-            let placeholderWithoutTry = placeholder.replacingOccurrences(of: "^try ", with: "", options: .regularExpression)
-            modifiersAndPlaceholder = "try await \(placeholderWithoutTry)"
-          } else {
-            modifiersAndPlaceholder = "try await \(placeholder)"
-          }
-          printer.print("let swiftResult$ = \(modifiersAndPlaceholder)")
+          // Make sure try/await are printed when necessary and avoid duplicate, or wrong-order, keywords (which would cause warnings)
+          let placeholderWithoutTry = 
+            if placeholder.hasPrefix("try ") {
+              String(placeholder.dropFirst("try ".count))
+            } else {
+              placeholder
+            }
+
+          let tryAwaitString: String = 
+            if isThrowing {
+              "try await"
+            } else {
+              "await"
+            }
+          printer.print("let swiftResult$ = \(tryAwaitString) \(placeholderWithoutTry)")
           printer.print("environment = try! JavaVirtualMachine.shared().environment()")
           let inner = nativeFunctionSignature.result.conversion.render(&printer, "swiftResult$")
           if swiftFunctionResultType.isVoid {

--- a/Tests/JExtractSwiftTests/JNI/JNIAsyncTests.swift
+++ b/Tests/JExtractSwiftTests/JNI/JNIAsyncTests.swift
@@ -144,7 +144,7 @@ struct JNIAsyncTests {
                   deferEnvironment.interface.DeleteGlobalRef(deferEnvironment, globalFuture)
                 }
                 do {
-                  let swiftResult$ = await try SwiftModule.async()
+                  let swiftResult$ = try await SwiftModule.async()
                   environment = try! JavaVirtualMachine.shared().environment()
                   _ = environment.interface.CallBooleanMethodA(environment, globalFuture, _JNIMethodIDCache.CompletableFuture.complete, [jvalue(l: nil)])
                 }
@@ -164,7 +164,7 @@ struct JNIAsyncTests {
                 deferEnvironment.interface.DeleteGlobalRef(deferEnvironment, globalFuture)
               }
               do {
-                let swiftResult$ = await try SwiftModule.async()
+                let swiftResult$ = try await SwiftModule.async()
                 environment = try! JavaVirtualMachine.shared().environment()
                 _ = environment.interface.CallBooleanMethodA(environment, globalFuture, _JNIMethodIDCache.CompletableFuture.complete, [jvalue(l: nil)])
               }

